src/braket/ir/jaqcd/instructions.py

```diff
@@ -899,3 +900,40 @@ class Type(str, Enum):
         kraus = "kraus"
 
     type = Type.kraus
+
+    """
+    StartPreserveBLock.
+
+    Attributes:
+        type (str): The instruction type. default = "start_preserve_block". (type) is optional.
+            This should be unique among all instruction types.
+
+    Examples:
+        >>> StartPreserveBlock(directive=["StartPreserveBlock"])
+    """
+
+class StartPreserveBlock(CompilerDirective):
+
+    class Type(str, Enum):
+        start_preserve_block = "start_preserve_block"
+
+    type = Type.start_preserve_block
+
+
+class EndPreserveBlock(CompilerDirective):
+    """
+    EndPreserveBLock.
```

Comment:
Please provide a description of the functionality.

Reply:
Updated